### ISSUE TITLE
low performance

### ISSUE THREAD

**Issue by physxP** [kkAyataka/plusaes#30]
I have been experiencing very low performance with the library when compared to openssl.

Specs:
mac m1

Algorithm: aes gcm 128 decryption
File size: 10mb
Plusaes Total time: 3 seconds > (with compiler optimizations) 
OpenSSL time: 2milliseconds
I have also tested the file on other platforms like android and ios and there I get 2-3 ms for decryption using native APIs.

Due to this I am unable to use this inside my project. Can I do anything to speed up the decryption process as this library is incredibly useful for crossplatform development?

**Comment by kkAyataka** [comment]
Hm,

- I think so, too (plusaes is slow) 
- Does not OpenSSL use hardware acceleration?
- In this library, Readability > Speed
    - Speeding up is not very important to me
   
It's not that I'm not interested in performance, but it's too much for me.

**Comment by physxP** [comment]
Sorry, I missed the Readability part and you are right about OpenSSL. If we try hardware acceleration and inline assembly then the library would face same problems as openssl in cross platform development. Another library that I found that gives good cross platform compatibility while being light weight is mbedtls so I have shifted decryption of large file on that library. This still works pretty good for small messages.
